### PR TITLE
Feature/241 improve handling of binary types in oracle dialect

### LINKED ISSUE #241: Improve handling of binary types in Oracle dialect
# Situation

Exasol has no binary types, Oracle does. At the moment the Oracle SQL dialect implemented in the Virtual Schema JDBC Adapter translates binary data via `TO_CHAR` function.
This has several problems:

- Binary data is not character data - even UTF-8 does not accept all possible bit combinations that can appear in a true binary column
- Exasol's `VARCHAR` column can only hold [up to 2*10^6 characters](https://docs.exasol.com/sql_references/data_types/datatypedetails.htm#StringDataType).
- Size limits inside conversion functions and the Oracle JDBC driver apply

### Ignoring Binary Columns

The simplest possible solution is to ignore binary columns in Virtual Schemas. In most cases this will be an acceptable option since especially large binaries are either images or documents, in rare cases other media like videos. None of them are particularly common in typical analytical RDBMs use cases.

### Converting Binary Columns to Base64 Encoded Text

Without modification binary data neither fits into UTF-8 nor ASCII strings. Encoding the data using the [Base64 algorithm](https://de.wikipedia.org/wiki/Base64) is a typical solution. Base64 encoding is very fast and not CPU-intensive. On the other hand three bytes binary require an four characters in the encoded form, adding an overhead of 33% on top of the actual data payload.

Given the maximum size of 2,000,000 characters for a `VARCHAR` column, we can encode up to 2 / 4 * 3 * 10 ^ 6 bytes of binary data this way. That means 1.5 * 10^6 bytes.

Size limits of the conversion algorithms used by Oracle additionally apply.

# Acceptance Criteria

- <strike>Users can choose between ignoring binary columns or having the data encoded with Base64</strike>
- By default binary columns are ignored
- <strike>Binary data that is too large to fit into an Exasol `VARCHAR` column is truncated before Base64 encoding</strike>

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #241.

### CODE REVIEW

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java [hunk 1]
```diff
@@ -19,6 +19,9 @@
  * Abstract implementation of a dialect. We recommend that every dialect should extend this abstract class.
  */
 public abstract class AbstractSqlDialect implements SqlDialect {
+    protected static final List<String> SHARED_PROPERTIES = Arrays.asList(SQL_DIALECT_PROPERTY,
```
Comment (AnastasiiaSergienko) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]: I'm not sure about the name. Maybe COMMON_PROPERTIES or STANDARD_PROPERTIES would be better?

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java [hunk 2]
```diff
@@ -150,13 +153,18 @@ protected void validateSupportedPropertiesList() throws PropertyValidationExcept
...
     }
 
+    /**
+     * Get the list of user-defined adapter properties which the dialect supports
+     *
+     * @return list of supported properties
+     */
+    protected abstract List<String> getSupportedProperties();
+
     protected String createUnsupportedElementMessage(final String unsupportedElement, final String property) {
```
Comment (AnastasiiaSergienko) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]: Unsupported Element? It's only about properties, isn't it?

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java (hunk 1 above)
Comment (redcatbear) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]: Actually that is the remnant of an experiment. I am going to remove that line. With Java 9 there might be a more elegant way to express the relationship between common and dialect-specific properties.

We should review all immutable lists with the Java 9 port anyway.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java (hunk 2 above)
Comment (redcatbear) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]: Check lines 186 and 193 to see the meaning.

**Pull request merged**